### PR TITLE
Integrate Redux for managing sidebar state

### PR DESCRIPTION
**Pull request by priyankatechneai**
Integrate Redux for managing sidebar state

### CODE REVIEW

File: src/components/Common/Sidebar.js [hunk 1]
```diff
@@ -1,346 +1,328 @@
...
+
+  const [isSidebarMini, setIsSidebarMini] = useState(false);
+
+  const [darkLightMode, setDarkLightMode] = useState('light');
+  const [updateRtl, setUpdateRtl] = useState(false);
+  const [user_id, setUser_id] = useState(localStorage.getItem('id'));
+  const [role_id, setRole_id] = useState(localStorage.getItem('role_id'));
+  const [account_for, setAccount_for] = useState('');
+
+  useEffect(() => {
```
Comment (somyaranjan-techneai) [inline src/components/Common/Sidebar.js:32]: why life call in  top

File: src/components/Common/Sidebar.js [hunk 2]
```diff
@@ -1,346 +1,328 @@
...
+  getMenuListThunk
+} from '../../redux/services/Sidebar';
+
+const Sidebar = ({ activekey }) => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const { menuList, filterMenuList } = useSelector((state) => state?.sidebar);
+
+  const menuListData = menuList?.menu;
```
Comment (somyaranjan-techneai) [inline src/components/Common/Sidebar.js:20]: please follow state sequency role

Comment (priyankatechneai) [reply src/components/Common/Sidebar.js:20]: resolved

File: src/components/Common/Sidebar.js (hunk 1 above)
Comment (priyankatechneai) [reply src/components/Common/Sidebar.js:32]: resolved

File: src/components/Common/Sidebar.js [hunk 3]
```diff
@@ -1,346 +1,327 @@
...
+
+  const [isSidebarMini, setIsSidebarMini] = useState(false);
 
- 
+  const [darkLightMode, setDarkLightMode] = useState('light');
+  const [updateRtl, setUpdateRtl] = useState(false);
+  const [user_id, setUser_id] = useState(localStorage.getItem('id'));
+  const [role_id, setRole_id] = useState(localStorage.getItem('role_id'));
+  const [account_for, setAccount_for] = useState('');
+  const { menuList, filterMenuList } = useSelector((state) => state?.sidebar);
```
Comment (somyaranjan-techneai) [inline src/components/Common/Sidebar.js:25]: state sequency should be in proper way and add comment also

File: src/components/Common/Sidebar.js [hunk 4]
```diff
@@ -1,346 +1,327 @@
...
-    } finally {
-    }
-  }
+    return JSON.parse(jsonPayload);
+  };
 
-  openChildren (id) {
-    var otherTabs = document.getElementsByClassName('has-children')
+  const openChildren = (id) => {
+    var otherTabs = document.getElementsByClassName('has-children');
```
Comment (somyaranjan-techneai) [inline src/components/Common/Sidebar.js:47]: use let

File: src/components/Common/Sidebar.js [hunk 5]
```diff
@@ -1,346 +1,327 @@
...
-        menutab.parentElement.children[0].setAttribute('aria-expanded', 'true')
+        menutab.parentElement.children[0].setAttribute('aria-expanded', 'true');
       }
     }
-  }
-  closeChildren () {
-    var otherTabs = document.getElementsByClassName('has-children')
+  };
+
+  const closeChildren = () => {
```
Comment (somyaranjan-techneai) [inline src/components/Common/Sidebar.js:99]: if you not use func why decal it

File: src/redux/slices/Sidebar/index.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      })
+      .addCase(getMenuListThunk.rejected, (state, action) => {
+        state.isLoading.getMenuList = false;
+        state.menuList = [];
+        state.errorMsg.getMenuList = action.error.message;
+      });
+  }
+});
+
+// export const {
```
Comment (somyaranjan-techneai) [inline src/redux/slices/Sidebar/index.js:65]: what is this

File: src/components/Common/Sidebar.js [hunk 7]
```diff
@@ -1,346 +1,327 @@
...
+const Sidebar = ({ activekey }) => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const [isSidebarMini, setIsSidebarMini] = useState(false);
 
- 
+  const [darkLightMode, setDarkLightMode] = useState('light');
+  const [updateRtl, setUpdateRtl] = useState(false);
+  const [user_id, setUser_id] = useState(localStorage.getItem('id'));
```
[the remark this answers is not in the document]
Comment (priyankatechneai) [reply src/components/Common/Sidebar.js:22]: resolved

File: src/components/Common/Sidebar.js (hunk 4 above)
Comment (priyankatechneai) [reply src/components/Common/Sidebar.js:47]: resolved

File: src/redux/slices/Sidebar/index.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+import { getEmployeeListThunk, getMenuListThunk } from '../../services/Sidebar';
+
+const initialState = {
+  employeeList: '',
```
[the remark this answers is not in the document]
Comment (priyankatechneai) [reply src/redux/slices/Sidebar/index.js:6]: resolved

File: src/components/Common/Sidebar.js (hunk 3 above)
Comment (priyankatechneai) [reply src/components/Common/Sidebar.js:25]: resolved

File: src/redux/slices/Sidebar/index.js (hunk 6 above)
Comment (priyankatechneai) [reply src/redux/slices/Sidebar/index.js:65]: resolved

File: src/redux/slices/Sidebar/index.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        state.isLoading.getEmployeeList = false;
+        state.employeeList = [];
+        state.errorMsg.getEmployeeList = action.error.message;
+      })
+
+      .addCase(getMenuListThunk.pending, (state, action) => {
+        state.isLoading.getMenuList = true;
+      })
+      .addCase(getMenuListThunk.fulfilled, (state, action) => {
+        state.isLoading.getMenuList = false;
```
[the remark this answers is not in the document]
Comment (priyankatechneai) [reply src/redux/slices/Sidebar/index.js:45]: resolved

File: src/components/Common/Sidebar.jsx [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+  const user_id = localStorage.getItem('id');
+  const role_id = localStorage.getItem('role_id');
+  //Redux State
+  const { menuList, filterMenuList } = useSelector((state) => state?.sidebar);
+
+  const menuListData = menuList?.menu;
+
+  //local state
+
+  const [menuData, setMenuData] = useState(filterMenuList?.[0]);
```
Comment (somyaranjan-techneai) [inline src/components/Common/Sidebar.jsx:27]: remove extra space

File: src/components/Common/Sidebar.jsx [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+          <span className="ms-2">
+            <i className="icofont-bubble-right"></i>
+          </span>
+        </button>
+      </div>
+    </div>
+  );
+};
+
+export default Sidebar;
```
Comment (somyaranjan-techneai) [inline src/components/Common/Sidebar.jsx:300]: please wrap in memo HOC

File: src/components/Common/Sidebar.jsx (hunk 10 above)
Comment (priyankatechneai) [reply src/components/Common/Sidebar.jsx:27]: resolved

File: src/components/Common/Sidebar.jsx (hunk 11 above)
Comment (priyankatechneai) [reply src/components/Common/Sidebar.jsx:300]: resolved

**Pull request merged**